conan/tools/files/files.py

```diff
@@ -520,3 +522,22 @@ def swap_child_folder(parent_folder, child_folder):
     child = os.path.join(parent_folder, child_folder)
     for f in os.listdir(child):
         shutil.move(os.path.join(child, f), os.path.join(parent_folder, f))
+
+
+def fix_apple_shared_install_name(conanfile):
+    def _osx_collect_dylibs(lib_folder):
+        return [f for f in os.listdir(lib_folder) if f.endswith(".dylib")
+                and not os.path.islink(os.path.join(lib_folder, f))]
+
+    def _fix_install_name(lib_name, lib_folder):
+        command = "install_name_tool -id @rpath/{} {}".format(lib_name, os.path.join(lib_folder,
+                                                                                     lib_name))
+        conanfile.run(command)
+
+    if is_apple_os(conanfile.settings.get_safe("os")) and conanfile.options.get_safe("shared", False):
```

Comment:
I think there could be situations (perhaps in private recipes) where the `shared` option is not defined (maybe the library can "only" work as a shared library and the maintainers didn't bother to add shared True/False toggle - can be the case with libraries that define global symbols were it can be dangerous if they end up in multiple shared libraries loaded by the same process..) 

I would probably err on the side of caution and not require that the `shared` option exists or is set to true? / Or perhaps, change the default in `get_safe` to true: assume that if it is not defined but the user is calling it, that they want this to be executed?

Reply:
This might be partially addressed in Conan 2.0 ``package_type`` which is the way to specify something is a shared library, even if no ``shared`` option is defined (because the package is always shared)